### DISCUSSION TITLE
Change the default Tauri application URL

### DISCUSSION THREAD

**Discussion by csindile-panaya** [tauri-apps/tauri#9174] [General]
It's possible to change the default Tauri application URL that is https://tauri.localhost/? If the answer is yes, how can we do that?

**Comment by FabianLars** [comment]
If you mean just the url, but not the actual behavior (== you still want tauri to host the distDir files), then no. You can only switch between http and https via https://tauri.app/v1/api/config/#securityconfig.dangeroususehttpscheme. Note that this is windows only, on macos and linux it's always tauri://localhost

If you want to change the url to some other serving mechanism you can either use the url property on the window configs or set devPath and distDir in tauri.conf.json to apply it to every window by default.

**Reply by csindile-panaya** [reply]
Hi @FabianLars,

Thanks for your response.
I will explain what I want to solve and maybe you can help me.
I have a user activity tracking script that is injected in my default page of the Tauri application. If I'm adding the value of "test" for the "url" property from "windows" object inside the tauri.conf.json file, the script will record that the default page URL of the Tauri application is "https://tauri.localhost/test" or "https://tauri.localhost/" like before? I would like to "fool" the injected script to get the default page URL of my Tauri application to be something like "https://tauri.localhost/appId" because I will have more than one Tauri application on the machine (I would like to differentiate them and get proper user activity tracking URLs for each application).
There is any solution to achieve this in a Tauri application?

Regards,
Claudiu

**Reply by FabianLars** [reply]
Hmm, so Tauri itself doesn't have anything explicitly for this, as it really only serves the distDir files as they are, _however_ it also falls back automatically to index.html to support SPA frameworks/routers. This won't change the origin to the fall back from my testing.
So for example, i openened my app, ran `window.location.replace("/appid")` and even though it serves /index.html the origin will indeed be `https://tauri.localhost/appId`.

This probably could be done a bit fancier with frontend bundler configs but i'm not an expert there (and of course it depends on your stack).

**Reply by csindile-panaya** [reply]
Thanks a lot @FabianLars !
I hope that will help other people also.

**Comment by liudonghua123** [comment]
I have a similar requirement which need to config the response header to support [SharedArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer), see https://webcontainers.io/guides/quickstart#cross-origin-isolation.

I tried to use `tauri-plugin-localhost` plugin which supports adding `Cross-Origin-Opener-Policy: same-origin` and `Cross-Origin-Embedder-Policy: require-corp` and serve the static files via http://localhost:<some_port> which is needed to enable [SharedArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer#security_requirements).

But the the builtin [tiny-http](https://github.com/tiny-http/tiny-http) lib does not support ranges features currently.

See also https://github.com/tauri-apps/plugins-workspace/issues/880.

**Reply by csindile-panaya** [reply]
@FabianLars , @liudonghua123,
Our application is developed in Angular with Tauri and we managed to add easily a '/appId' suffix to original URL just using a line of code:
{ provide: APP_BASE_HREF, useValue: '/appId' } inside the providers declaration in our application module.
With this change our production application URL will be https://tauri.localhost/appId/ which is exactly what we want to differentiate our applications.
Thanks for all your help and support.

**Comment by daveycodez** [comment]
This is required for password autofill from web to app
